cmd/cstor-pool-mgmt/controller/new-pool-controller/new_pool_controller.go

```diff
@@ -0,0 +1,198 @@
+	kubeclientset kubernetes.Interface
+
+	// clientset is a openebs custom resource package generated for custom API group.
+	clientset clientset.Interface
+
+	// cStorPoolSynced is used for caches sync to get populated
+	cStorPoolSynced cache.InformerSynced
+
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens. This
+	// means we can ensure we only process a fixed amount of resources at a
+	// time, and makes it easy to ensure we are never processing the same item
+	// simultaneously in two different workers.
+	workqueue workqueue.RateLimitingInterface
+	// recorder is an event recorder for recording Event resources to the
+	// Kubernetes API.
+	recorder record.EventRecorder
+}
+
+// NewCStorPoolController returns a new instance of CStorPool controller
+func NewCStorPoolController(
+	kubeclientset kubernetes.Interface,
+	clientset clientset.Interface,
+	kubeInformerFactory kubeinformers.SharedInformerFactory,
+	cStorInformerFactory informers.SharedInformerFactory) *CStorPoolController {
+
+	// obtain references to shared index informers for the cStorPool resources
+	cStorPoolInformer := cStorInformerFactory.Openebs().V1alpha2().CStorNPools()
+
+	pool.KubeClient = kubeclientset
+	pool.OpenEBSClient2 = clientset
+	pool.ImportedCStorPools = map[string]*apis2.CStorNPool{}
+
+	err := openebsScheme.AddToScheme(scheme.Scheme)
+	if err != nil {
+		glog.Errorf("failed to add to scheme: error {%v}", err)
+		return nil
+	}
+
+	// Create event broadcaster to receive events and send them to any EventSink, watcher, or log.
+	// Add NewCstorPoolController types to the default Kubernetes Scheme so Events can be
+	// logged for CstorPool Controller types.
+	glog.V(4).Info("Creating event broadcaster")
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+
+	// StartEventWatcher starts sending events received from this EventBroadcaster to the given
+	// event handler function. The return value can be ignored or used to stop recording, if
+	// desired. Events("") denotes empty namespace
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: poolControllerName})
+
+	controller := &CStorPoolController{
+		kubeclientset:   kubeclientset,
+		clientset:       clientset,
+		cStorPoolSynced: cStorPoolInformer.Informer().HasSynced,
+		workqueue:       workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "CStorNPool"),
+		recorder:        recorder,
+	}
+
+	glog.Info("Setting up event handlers for CSP")
+
+	// Set up an event handler for when CstorPool resources change.
+	cStorPoolInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			csp := obj.(*apis2.CStorNPool)
+			if !pool.IsRightCStorPoolMgmt(csp) ||
```

Comment:
checking for status doesn't make sense.. lets handle it in the handler code
Lets just check for RightCStorPoolMgmt and disableReconcilation

Reply:
ok